src/mixins/action.js

```diff
@@ -0,0 +1,60 @@
+
+/**
+ * @copyright Copyright (c) 2019 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+export default {
+	props: {
+		icon: {
+			type: String,
+			default: '',
+			required: true
+		},
+		title: {
+			type: String,
+			default: ''
+		}
+	},
+
+	computed: {
+		isIconUrl() {
+			try {
+				return new URL(this.icon)
+			} catch (error) {
+				return false
+			}
+		},
+		text() {
+			return this.$slots.default[0].text
```

Comment:
this will :boom: if you forget to fill the slot. Shall we warn in that case?

Reply:
@ChristophWurst nice idea yes!!

Reply:
Done